test/e2e_node/podresources_test.go

```diff
@@ -702,6 +702,26 @@ var _ = SIGDescribe("POD Resources [Serial] [Feature:PodResources][NodeFeature:P
 					podresourcesListTests(ctx, f, cli, nil)
 					podresourcesGetAllocatableResourcesTests(ctx, cli, nil, onlineCPUs, reservedSystemCPUs)
 				})
+				ginkgo.It("should account for resources of pods in terminal phase", func(ctx context.Context) {
+					pd := podDesc{
+						cntName:    "e2e-test-cnt",
+						podName:    "e2e-test-pod",
+						cpuRequest: 1000,
+					}
+					pod := makePodResourcesTestPod(pd)
+					pod.Spec.Containers[0].Command = []string{"sh", "-c", "sleep 10s"}
+					pod = e2epod.NewPodClient(f).CreateSync(ctx, pod)
+					err := e2epod.WaitForPodCondition(ctx, f.ClientSet, pod.Namespace, pod.Name, "Pod Succeeded", time.Minute*2, testutils.PodSucceeded)
+					framework.ExpectNoError(err)
+					endpoint, err := util.LocalEndpoint(defaultPodResourcesPath, podresources.Socket)
+					framework.ExpectNoError(err)
+					cli, conn, err := podresources.GetV1Client(endpoint, defaultPodResourcesTimeout, defaultPodResourcesMaxSize)
+					framework.ExpectNoError(err)
+					defer conn.Close()
+					// although the pod moved into terminal state, PodResourcesAPI still list its cpus
+					expectPodResources(ctx, 1, cli, []podDesc{pd})
+					e2epod.NewPodClient(f).DeleteSync(ctx, pod.Name, metav1.DeleteOptions{}, time.Minute)
```

Comment:
this should be probably moved into a `defer` or hooked into `AfterEach` (yes, the surrounding tests are inconsistent/sometimes wrong)

Reply:
@ffromani Thanks, I used `defer` in this case since the `AfterEach` would apply to all the subject nodes (`It`'s) in the spec which is not necessarily logically true.